src/dataconnect/fileUtils.ts

```diff
@@ -137,3 +137,38 @@ export async function directoryHasPackageJson(dirPath: string) {
   const fileNames = await fs.readdir(dirPath);
   return fileNames.some((f) => f.toLowerCase() === "package.json");
 }
+
+// Generates sdk yaml based on platform, and returns a modified connectorYaml
+export function generateSdkYaml(
+  platform: Platform,
+  connectorYaml: ConnectorYaml,
+  connectorYamlFolder: string, // path.relative expects folder as first arg
+  appFolder: string,
+): ConnectorYaml {
+  const relPath = path.relative(connectorYamlFolder, appFolder);
+  const outputDir = path.join(relPath, "dataconnect-generated");
+  if (!connectorYaml.generate) {
+    connectorYaml.generate = {};
+  }
+  if (platform === Platform.WEB) {
+    connectorYaml.generate.javascriptSdk = {
+      outputDir,
+      package: `@firebasegen/${connectorYaml.connectorId}`,
+      packageJsonDir: appFolder,
```

Comment:
Edge case to check for when using this function - If i don't provide an app folder, what does this write to connector.yaml? Does it leave an empty `packageJsonDir:`?

Reply:
hrm, are you suggesting we check if appFolder is empty? outputDir is also being generated from appFolder

Reply:
This is looking forward to follow up PRs that start using this function, so no need to block on this: 'packageJsonDir' is optional, to support users that use package managers other than npm. We should make sure that the UI makes it easy for users to omit it, and we should make sure that the final `connector.yaml` doesn't have an empty `packageJsonDir` field:

```
generate:
 webSdk:
    outputDir: ../../dir
    package: connectors.default
    packageJsonDir:
```

Reply:
ah, this would be for CLI init flow no? The auto detect doesn't have any web indicators aside from npm indiciators.